pkg/skaffold/filemon/monitor_test.go

```diff
@@ -29,76 +26,59 @@ import (
 func TestFileMonitor(t *testing.T) {
 	var tests = []struct {
 		description string
-		update      func(folder *testutil.TempDir)
+		makeChanges func(folder *testutil.TempDir)
 	}{
 		{
 			description: "file change",
-			update: func(folder *testutil.TempDir) {
+			makeChanges: func(folder *testutil.TempDir) {
 				folder.Chtimes("file", time.Now().Add(2*time.Second))
 			},
 		},
 		{
 			description: "file delete",
-			update: func(folder *testutil.TempDir) {
+			makeChanges: func(folder *testutil.TempDir) {
 				folder.Remove("file")
 			},
 		},
 		{
 			description: "file create",
-			update: func(folder *testutil.TempDir) {
-				folder.Write("new", "content")
+			makeChanges: func(folder *testutil.TempDir) {
+				folder.Touch("new")
 			},
 		},
 	}
 	for _, test := range tests {
 		testutil.Run(t, test.description, func(t *testutil.T) {
-			tmpDir := t.NewTempDir().
-				Write("file", "content")
+			tmpDir := t.NewTempDir().Touch("file")
 
-			folderChanged := newCallback()
-
-			// Watch folder
 			monitor := NewMonitor()
-			err := monitor.Register(tmpDir.List, folderChanged.call)
-			t.CheckNoError(err)
 
-			// Run the watcher
-			ctx, cancel := context.WithCancel(context.Background())
-			var stopped sync.WaitGroup
-			stopped.Add(1)
-			go func() {
```

Comment:
Nice, that's much simpler.

It really makes more sense to test synchronous code with a synchronous test :)